### PR TITLE
Optional inputs handling in the ONNX tests runner

### PR DESCRIPTION
**Pull request by tomdol**
### Details:
 - The PR removes empty input values provided by the testing framework. Those empty values (empty numpy arrays) are passed in as placeholders for optional inputs of an operator. This happens for example in the Resize op test that was previously xfailed.
 - With those changes the test still fails but for another reason - the CPU plugin receives a dynamic Interpolate op. This is why the test remains xfailed but the marker (and ticket) has changed.

### Tickets:
 - 38714
 - 49750

### CODE REVIEW

File: ngraph/python/src/ngraph/utils/types.py [hunk 1]
```diff
@@ -133,6 +133,24 @@ def get_shape(data: NumericData) -> TensorShape:
...
 
+def is_empty_array(obj: Any) -> bool:
+    """Return true if a given object is an empty numpy array."""
+    if type(obj) == np.ndarray:
+        return obj.size == 0
+    else:
+        return False
+
+
+def remove_empty_inputs(inputs: List[NumericData], params_number: int) -> List[NumericData]:
```
Comment (postrational) [inline ngraph/python/src/ngraph/utils/types.py:144]: We should move this helper into `ngraph/python/tests`

**Review by tsocha — COMMENTED** [review]
I don't like these changes, it looks like a workaround to pass a test which is not correct from the ONNX spec perspective.
It is similar to #4340 but there was a big model which was affected by it.

**Comment by tomdol** [comment]
> I don't like these changes, it looks like a workaround to pass a test which is not correct from the ONNX spec perspective.
> It is similar to #4340 but there was a big model(onnx model zoo) which was affected by it.

How about merging it now and reverting if/when the test is fixed in the ONNX repo?

**Comment by tomdol** [comment]
Test model adjusted in the ONNX repo. Changes will be available in release 1.9

**Pull request closed without merging**